test/integration/dynamic_repo_test.exs

```diff
@@ -0,0 +1,101 @@
+defmodule Oban.Integration.DynamicRepoTest do
+  use Oban.Case
+  import Ecto.Query
+  alias Oban.Test.DynamicRepo
+
+  setup do
+    {:ok, repo_pid} = start_supervised({DynamicRepo, name: nil})
+
+    DynamicRepo.put_dynamic_repo(repo_pid)
+    Repo.delete_all(Job)
+    Repo.delete_all(Job, prefix: "private")
+    DynamicRepo.put_dynamic_repo(nil)
+
+    on_exit(fn ->
+      DynamicRepo.put_dynamic_repo(repo_pid)
+      Repo.delete_all(Job)
+      Repo.delete_all(Job, prefix: "private")
+    end)
+
+    {:ok, repo_pid: repo_pid}
+  end
+
+  test "job execution", context do
+    name = start_oban!(context.repo_pid, queues: [alpha: 1])
+    job_ref = insert_job!(name).ref
+    assert_receive {:ok, ^job_ref}
+  end
+
+  test "draining", context do
```

Comment:
The `Drainer` module uses the same `Query` calls as a queue producer and this may not exercise any additional `Repo` calls. Do you think we need it as an additional case?

Reply:
I wasn't sure about the internals, so I decided to include it in tests. If you think it's redundant let me know and I'll remove it.